lib/workload/stateless/stacks/bclconvert-manager/translator_service/icav2_event_translator.py

```diff
@@ -0,0 +1,504 @@
+def store_events_into_dynamodb(internal_ica_event, table_name, event_details) -> None:
+    try:
+        db_uuid = generate_db_uuid(None, None).get("db_uuid",'')
+        dynamodb.put_item(
+            TableName=table_name,
+            Item={
+                'id': {'S': db_uuid},
+                'id_type': {'S': 'db_uuid'},
+                'analysis_id': {'S': internal_ica_event.detail.payload.get('data', '').get("analysisId", '')},
+                'analysis_status': {'S': internal_ica_event.detail.status},
+                # 'SUCCEEDED', 'FAILED', 'INPROGRESS', 'ABORTED', 'UNKNOWN
+                "portal_run_id": {'S': internal_ica_event.detail.portalRunId},
+                'original_external_event': {'S': json.dumps(event_details)},
+                'translated_internal_ica_event': {'S': json.dumps(internal_ica_event.detail.to_dict())},
+                'timestamp': {'S': internal_ica_event.detail.timestamp}
+            }
+        )
+        logger.info(f"Original and Internal events stored in the DynamoDB table.")
+
+        # update dynamodb table with the new generate db_uuid
+        dynamodb.update_item(
+            TableName=table_name,
+            Key={
+                'id': {'S': internal_ica_event.detail.payload.get("data", '').get("analysisId", '')},
+                'id_type': {'S': 'analysis_id'}
+            },
+            UpdateExpression='SET db_uuid = :db_uuid',
+            ExpressionAttributeValues={
+                ':db_uuid': {'S': db_uuid}
+            }
+        )
+        dynamodb.update_item(
+            TableName=table_name,
+            Key={
+                'id': {'S': internal_ica_event.detail.portalRunId},
+                'id_type': {'S': 'portal_run_id'}
+            },
+            UpdateExpression='SET db_uuid = :db_uuid',
+            ExpressionAttributeValues={
+                ':db_uuid': {'S': db_uuid}
+            }
+        )
+        logger.info(f"db_uuid updated in anaylsis_id and portal_run_id record.")
+    except Exception as e:
+        raise Exception("Failed to store event in the DynamoDB table. Error: ", e)
+
+
+# Convert from Entity model to aws event object with aws event envelope
+def translate_to_aws_event(event) -> AWSEvent:
+    return AWSEvent(
+        detail=get_event_details(event),
+        detail_type="WorkflowRunStateChange",
+        source="orcabus.bclconvertmanager",
+        # version="0.1.0",  # comment as the version is managed by the evnet bus
+    )
+
+
+# Convert from entity module to internal event details
+def get_event_details(event) -> WorkflowRunStateChange:
+    # Extract relevant fields from the event payload
+    analysis_status = event.get("eventParameters", {}).get("analysisStatus", '')
+
+    payload = event.get("payload", {})
+    analysis_id = payload.get("id", '')
+    pipeline = payload.get("pipeline", {})
+
+    event_name, version = parse_event_code(pipeline.get("code", ''))
+    
+    if analysis_status != "SUCCEEDED":
```

Comment:
An empty payload for all non-SUCCEEDED events is fine for now.
In the future we may want to handle all known statuses, for example add an error message payload for FAILED events. Happy to put that on the backlog though.

Reply:
sure, new backlog added [here](https://github.com/orgs/umccr/projects/21/views/1?pane=issue&itemId=65106537)
I think for non-final status "Requested", "Queued", "Initializing", "Preparing Inputs", "In Progress", "Generating outputs", "Aborting", we can leave it empty;
for final status "Aborted", "Failed", "Succeeded", we may need add additional specific information in the future.